hs_core/hydroshare/utils.py

```diff
@@ -459,6 +463,8 @@ def copy_resource_files_and_AVUs(src_res_id, dest_res_id, set_to_private=False):
         if value:
             if avu_name == 'isPublic' and set_to_private:
                 istorage.setAVU(dest_coll, avu_name, 'False')
+            elif avu_name == 'quotaUserName' and requesting_user:
+                istorage.setAVU(dest_coll, avu_name, requesting_user.username)
```

Comment:
@alvacouch The reason to call copy_resource_files_and_AVUs after an empty resource is created is to accommodate the scenario where copy_resouece_files_and_AVUs failed in which case the empty resource is deleted, so empty resource creation is a precursor for copy_resource_files_and_AVUs for resource copy workflow and they are atomic in that copy_resource_files_and_AVUs always happens after the empty resource is created if both operations are successful, and if either operation is not successful, the workflow will roll back so no resource is created. So the quota will be set only when both create empty resource and copy_resource_files_and_AVUs are successful. Does that make sense to you?